plugwise/stick.py

```diff
@@ -498,769 +375,179 @@ def _append_node(self, mac, address, node_type):
-                                        1
-                                    ].__class__.__name__
-                                ),
-                            )
-                            self.send(
-                                self.expected_responses[seq_id][1],
-                                self.expected_responses[seq_id][2],
-                                self.expected_responses[seq_id][3] + 1,
-                            )
-                        else:
-                            if isinstance(
-                                self.expected_responses[seq_id][1], NodeAddRequest
-                            ) or isinstance(
-                                self.expected_responses[seq_id][1], StickInitRequest
-                            ):
-                                _LOGGER.info(
-                                    "Drop %s request because max (%s) retries reached for seq id %s",
-                                    self.expected_responses[seq_id][
-                                        1
-                                    ].__class__.__name__,
-                                    str(MESSAGE_RETRY),
-                                    str(seq_id),
-                                )
-                            else:
-                                if self.expected_responses[seq_id][1].mac == "":
-                                    mac = "<empty>"
-                                else:
-                                    mac = self.expected_responses[seq_id][1].mac.decode(
-                                        UTF8_DECODE
-                                    )
-                                _LOGGER.info(
-                                    "Drop %s request for mac %s because max (%s) retries reached for seq id %s",
-                                    self.expected_responses[seq_id][
-                                        1
-                                    ].__class__.__name__,
-                                    mac,
-                                    str(MESSAGE_RETRY),
-                                    str(seq_id),
-                                )
-                        del self.expected_responses[seq_id]
-            receive_timeout_checker = 0
-            while (
-                receive_timeout_checker < MESSAGE_TIME_OUT
-                and self._run_receive_timeout_thread
-            ):
-                time.sleep(1)
-                receive_timeout_checker += 1
-        _LOGGER.debug("Receive timeout loop stopped")
+        _LOGGER.warning(
+            "Invalid mac '%s' address, unable to unjoin node manually.", mac
+        )
+        return False
 
-    def new_message(self, message: NodeResponse):  # noqa: C901
-        """ Received message from Plugwise Zigbee network """
-        # TODO: flake8 indicates scan is too complex, level 47! indenting is indeed overly complex
+    def _remove_node(self, mac):
+        """Remove node from list of controllable nodes."""
```

Comment:
If a mac isn't in the list, shouldn't we raise/log? (Just asking out loud)

Reply:
Fixed in [d33a5e9](https://github.com/plugwise/python-plugwise/pull/55/commits/d33a5e933a4fcbfc40bb78d614eef66c1fad1842)